woocommerce/woocommerce
# [CYS on core] Update the WooCommerce Extensions Themes page to include references to the CYS

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Here a series of changes are being made to the **WooCommerce Extensions Themes** page to include references to the new CYS flow, including:

- Add the Design your own button. If the current active theme is TT4, it directs users to the assembler, if not, the theme switch warning modal is displayed to confirm the user wants to proceed to the CYS flow and change their current active theme.
- Add the Design your own section right after the second row of themes. If the current active theme is TT4, the "Start designing" button directs users to the assembler; if not, the theme switch warning modal is displayed to confirm the user wants to proceed to the CYS flow and change their current active theme. It is also important to note that the position of the banner is dynamically changed on screen resize to ensure it preserves its expected position in the second row of themes independent of the number of columns displayed:

- Add the line Didn't find a theme you like? Browse the WordPress.org theme directory to discover more. The link sends the user to Appearance -> Themes -> Add new Theme with e-commerce as a search term.

Closes #45418.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have the WooCommerce Beta Tester plugin installed and activated
2. Under Tools > WCA Test Helper > Features, make sure `customize-store` is enabled
3. Under Tools > WCA Test Helper > Tools, click on `Reset Customize Your Store`
4. Head over to WooCommerce > Extensions > Themes `wp-admin/admin.php?page=wc-admin&tab=themes&path=%2Fextensions`
5. Make sure you can visualize the Design your own button. If the current active theme is TT4, it directs users to the assembler, if not, the theme switch warning modal is displayed to confirm the user wants to proceed to the CYS flow and change their current active theme:

<img width="1650" alt="Screenshot 2024-03-12 at 09 20 07" src="https://github.com/woocommerce/woocommerce/assets/15730971/0cd9a1bf-e66b-41cb-9261-3ed5cdebf886">

<img width="1629" alt="Screenshot 2024-03-12 at 09 21 56" src="https://github.com/woocommerce/woocommerce/assets/15730971/3f1eaa16-7bf8-4281-abee-a4a7e09a3bf5">
6. Make sure you can visualize the Design your own section right after the second row of themes. If the currently active theme is TT4, the "Start designing" button directs users to the assembler; if not, the theme switch warning modal is displayed to confirm the user wants to proceed to the CYS flow and change their current active theme:

<img width="1233" alt="Screenshot 2024-03-12 at 09 24 42" src="https://github.com/woocommerce/woocommerce/assets/15730971/f8d8fc4b-a06a-4e7e-90da-42b755e1aab1">

<img width="1227" alt="Screenshot 2024-03-12 at 09 26 25" src="https://github.com/woocommerce/woocommerce/assets/15730971/3173181a-b6c8-48d1-9a2c-db6e0bb28824">

It is also important to note that the position of the banner is dynamically changed on screen resize to ensure it preserves its expected position in the second row of themes independent of the number of columns displayed:

https://github.com/woocommerce/woocommerce/assets/15730971/747e19ee-e109-462f-9a3b-dbd6f2d67f23

7. Make sure you can visualize the line Didn't find a theme you like? Browse the WordPress.org theme directory to discover more. The link sends the user to Appearance -> Themes -> Add new Theme with e-commerce as a search term.

<img width="1672" alt="Screenshot 2024-03-12 at 09 27 18" src="https://github.com/woocommerce/woocommerce/assets/15730971/7b97f15b-30f2-4173-a601-c049313067ca">


### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Update the WooCommerce Extensions Theme page to include references to the Customize Your Store flow.

#### Comment 

</details>
